Documentation/gettingstarted/taints.rst

```diff
@@ -0,0 +1,87 @@
+   the taint is removed.
+
+2. Cilium runs on the node, initializes it and, once ready, removes the
+   ``node.cilium.io/agent-not-ready`` taint.
+
+3. From this point on, pods will start being scheduled and running on the node,
+   having their networking managed by Cilium.
+
+The taint's effect should be chosen taking into account the following
+considerations:
+
+* If ``NoSchedule`` is used, pods won't be *scheduled* to a node until Cilium
+  has the chance to remove the taint. However, one practical effect of this is
+  that if some external process (such as a reboot) resets the CNI configuration on
+  said node, pods that were already scheduled will be allowed to start
+  concurrently with Cilium when the node next reboots, and hence may become
+  unmanaged and have their networking being managed by another CNI plugin.
+
+* If ``NoExecute`` is used, pods won't be *executed* on a node until Cilium has
+  had the chance to remove the taint. One practical effect of this is that
+  whenever the taint is added back to the node by some external process (such as
+  during an upgrade or eventually a routine operation), pods will be evicted from
+  the node until Cilium has had the chance to remove the taint.
+
+Another important thing to consider is the concept of node itself, and the
+different point of views over a node. For example, the instance/VM which backs a
+Kubernetes node can be patched or reset filesystem-wise by a cloud provider, or
+altogether replaced with an entirely new instance/VM that comes back with the
+same name as the already-existing Kubernetes ``Node`` resource. Even though in
+said scenarios the node-pool-level taint will be added back to the ``Node``
+resource, pods that were already scheduled to the node having this name will run
+on the node at the same time as Cilium, potentially becoming unmanaged. This is
+why ``NoExecute`` is recommended, as assuming the taint is added back in this
+scenario, already-scheduled pods won't run.
+
+However, on some environments or cloud providers, and as mentioned above, it may
+happen that a taint established at the node-pool level is added back to a node
+after Cilium has removed it and for reasons other than a node upgrade/reset.
+The exact circumstances in which this may happen may vary, but this may lead to
+unexpected/undesired pod evictions in the particular case when ``NoExecute`` is
+being used as the taint effect. It is, thus, recommended that in each deployment
+and depending on the environment or cloud provider, a careful decision is made
+regarding the taint effect (or even regarding whether to use the taint-based
+approach at all) based on the information above, on the environment or cloud
+provider's documentation, and on the fact that one is essentially establishing
+a trade-off between having unmanaged pods in the cluster (which can lead to
+dropped traffic and other issues) and having unexpected/undesired evictions
+(which can lead to application downtime).
```

Comment:
Seems like this paragraph is essentially referencing https://github.com/cilium/cilium/issues/17631. Might be worth mentioning it.

Reply:
I'm not 100% sure — it may be in some situations, and it's a bit unpredictable because we don't have a lot of visibility into what's happening over time in a given cloud provider and across the different cloud providers, but the paragraph actually and essentially tries to address the case in which a taint is added back to a node, causing pods to be evicted/deleted. So I am actually not sure if, in all of these cases, the CEPs are kept.